pkg/operator/operator.go

```diff
@@ -113,17 +134,16 @@ func (c OpenShiftControllerManagerOperator) sync() error {
 		condition.Message = messageString
 		v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, condition)
 	}
-	if !equality.Semantic.DeepEqual(operatorConfig.Status, originalOperatorConfig.Status) {
-		if _, err := c.operatorConfigClient.OpenShiftControllerManagers().UpdateStatus(context.TODO(), operatorConfig, metav1.UpdateOptions{}); err != nil {
-			return err
-		}
+	if operatorConfig, err = c.updateStatus(originalOperatorConfig, operatorConfig); err != nil {
+		return err
```

Comment:
Worth asking since we are refactoring - do we need potentially two calls to `updateStatus` per sync loop?

Reply:
As per @gabemontero [comment](https://github.com/openshift/cluster-openshift-controller-manager-operator/pull/173#discussion_r488654426), we do need to empty out the conditions when going from one management state to another.

Note that the `updateStatus()` routine only issues the update when the status is different, so potentially we would only have two API calls issued when changing the management state.

Reply:
I don't understand why the deepequal was removed. Why call the server if you don't have to?

Reply:
@sttts the `DeepEqual` call has been moved here [operator.go:93](https://github.com/otaviof/cluster-openshift-controller-manager-operator/blob/83e1703c7acda4d705d108ca9f38a1a24ec2d0bd/pkg/operator/operator.go#L93).